CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc

```diff
@@ -1236,8 +1237,14 @@ static function HandlePostMissionPOI(XComGameState NewGameState, XComGameState_A
 				MissionState.PickPOI(NewGameState);
 				`CI_Warn(ActivityState.GetMyTemplateName() $ " has no POI and is marked as requiring one! Spawning replacement");
 			}
-		
-			class'X2StrategyElement_DefaultMissionSources'.static.SpawnPointOfInterest(NewGameState, MissionState);
```

Comment:
Instead of this hack (which will also not work for retals & other stuff that doesn't use chain system), you should instead call `XCGS_BM::ForceCloseBlackMarket` at the campaign setup and then "manually replicate" `ShowBlackMarket` method in the reward code